docs/overview.rst

```diff
@@ -1,20 +1,110 @@
 Why new programming language
 ============================
 
-Taichi is a high-performance programming language for computer graphics applications. The design goals are
+Taichi is a high-performance programming language for computer graphics applications.
+
+Introduction
+------------
+
+Different backends
+******************
+
+Before Taichi has been invented, to write a **general purposed GPU (GPGPU)**
+program, we have to manipulate through tons of different APIs, e.g.:
+
+``cuDeviceSynchronize()`` for CUDA, ``glDispatchCompute()`` for OpenGL,
+``pthread_join()`` for multi-core CPUs...
+
+Downside:
+
+1. CUDA for example, despite it's powerfulness, is only supported on NVIDIA
+   cards. One could not share/distribute his/her work with/to a computer
+   without NVIDIA GPUs.
+   This means the loss of portability.
+
+2. OpenGL for example, due to the historical debt, is too complicated when
+   used as a GPGPU programming tool for beginners, make people easily get
+   lost in API manuals and could not focus the real goal.
+   This could depress the productivity.
```

Comment:
```suggestion
2. OpenGL, due to its historical debt, is too complicated for beginners when
   used as a GPGPU programming tool. People could easily get
   lost in API manuals instead of focusing on their original goals.
   This could cause the loss of productivity.
```